src/Services/InvoiceLimitationsService.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Invoice\Services;
+
+use Invoice\Helper\SettingsHelper;
+use Plenty\Modules\Account\Contact\Models\Contact;
+use Plenty\Modules\Account\Contact\Models\ContactAllowedMethodOfPayment;
+use Plenty\Modules\Frontend\Contracts\CurrencyExchangeRepositoryContract;
+use Plenty\Modules\Order\Models\OrderAmount;
+
+/**
+ * Class InvoiceLimitationsService
+ *
+ * @author emmanouil.stafilarakis <emmanouil.stafilarakis@plentymarkets.com>
+ *
+ * @package Invoice\Services
+ */
+class InvoiceLimitationsService
+{
+    /**
+     * Get whether all payment method limitations are respected or not.
+     * 
+     * @param SettingsHelper $settingsHelper    The setting helper 
+     * @param bool           $isGuest           Whether or not the current customer is a guest or not
+     * @param OrderAmount    $amount            The total amount
+     * @param int            $shippingCountryId The ID of the shipping country
+     * @param int            $billingAddressId  The ID of the billing address
+     * @param int            $deliveryAddressId The ID of the delivery address
+     * @param Contact|null   $contact           The contact instance, if customer is not a quest
+     * 
+     * @return bool
+     */
+    public function respectsAllLimitations(
+        SettingsHelper $settingsHelper,
+        bool $isGuest,
+        OrderAmount $amount,
```

Comment:
Why did you add an OrderAmount Object as parameter here? Especially since where it is used, you are "faking" this OrderAmount-Object with only setting the total amount and the currency?

Why can't you just use two parameters, which makes it more explicite to read, seeing that both a float and string are given here?